language/move-lang/src/typing/core.rs

```diff
@@ -732,15 +749,31 @@ pub fn make_function_type(
     };
     let defined_loc = finfo.defined_loc;
     match &finfo.visibility {
-        FunctionVisibility::Internal if !in_current_module => {
-            let internal_msg = "This function is internal to its module. Only 'public' functions \
+        FunctionVisibility::Internal => {
+            if !in_current_module {
+                let internal_msg =
+                    "This function is internal to its module. Only 'public' functions \
```

Comment:
'public' and 'public(script)'? 

Then after friends we can change this to 'public' and 'public(_)'?